components/ILIAS/Mail/classes/class.ilMimeMail.php

```diff
@@ -266,60 +286,71 @@ protected function buildBodyMultiParts(string $skin, string $style): void
             $this->body = $this->refinery->string()->makeClickable()->transform(nl2br($this->body));
         } else {
             // if there is HTML, convert "<br>" to "\n" and strip tags for plain text alternative
-            $this->finalBodyAlt = strip_tags(str_ireplace(["<br />", "<br>", "<br/>"], "\n", $this->body));
+            $this->finalBodyAlt = strip_tags(str_ireplace(['<br />', '<br>', '<br/>'], "\n", $this->body));
         }
 
-        $this->finalBody = str_replace('{PLACEHOLDER}', $this->body, $this->getHtmlEnvelope($skin, $style));
+        return $this->refinery->string()->makeClickable()->transform(nl2br($body));
     }
 
-    private function getPathToRootDirectory(): string
+    private function getStyleSheetPath(string $skin, string $style): string
     {
-        return realpath(dirname(__DIR__, 4) . '/');
-    }
-
-    protected function getHtmlEnvelope(string $skin, string $style): string
-    {
-        $bracket_path = $this->getPathToRootDirectory() . '/components/ILIAS/Mail/templates/default/tpl.html_mail_template.html';
         if ($skin !== 'default') {
             $locations = [
                 $skin,
                 $skin . '/' . $style
             ];
 
             foreach ($locations as $location) {
-                $custom_path = $this->getPathToRootDirectory(
-                ) . '/public/Customizing/skin/' . $location . '/components/ILIAS/Mail/tpl.html_mail_template.html';
+                $custom_path = $this->getPathToRootDirectory() . "/public/Customizing/skin/$location/mail.css";
                 if (is_file($custom_path)) {
-                    $bracket_path = $custom_path;
-                    break;
+                    return $custom_path;
                 }
             }
         }
 
-        return file_get_contents($bracket_path);
+        return 'assets/css/mail.css';
     }
 
-    protected function buildHtmlInlineImages(string $skin, string $style): void
+    private function getLogoCid(string $skin, string $style): string
     {
-        $this->gatherImagesFromDirectory(
-            $this->getPathToRootDirectory() . '/components/ILIAS/Mail/templates/default/img'
-        );
-
         if ($skin !== 'default') {
             $locations = [
                 $skin,
                 $skin . '/' . $style
             ];
 
             foreach ($locations as $location) {
-                $custom_directory = $this->getPathToRootDirectory(
-                ) . '/public/Customizing/skin/' . $location . '/components/ILIAS/Mail/img';
+                $custom_directory = $this->getPathToRootDirectory() . "/public/Customizing/skin/$location/images/logo";
```

Comment:
Can we move this sub-path to a private PHP `string` constant (containing a placeholder for `$location`) to the top of the class and just replace the placeholder here?